lib/ansible/modules/cloud/vmware/vmware_vspan_session.py

```diff
@@ -0,0 +1,582 @@
+            wait_for_task(task)
+        except Exception:
+            self.restore_original_state()
+            self.module.fail_json(msg=task.info.error.msg)
+
+    def restore_original_state(self):
+        """In case of failure restore, the changes we made."""
+        for port, state in self.modified_ports.items():
+            self.set_port_security_promiscuous(port, state)
+        if self.deleted_session is not None:
+            session = self.deleted_session
+            config_version = self.dv_switch.config.configVersion
+            s_spec = vim.dvs.VmwareDistributedVirtualSwitch.VspanConfigSpec(vspanSession=session, operation="add")
+            c_spec = vim.dvs.VmwareDistributedVirtualSwitch.ConfigSpec(vspanConfigSpec=[s_spec], configVersion=config_version)
+            # Revert the delete
+            task = self.dv_switch.ReconfigureDvs_Task(c_spec)
+            try:
+                wait_for_task(task)
+            except Exception:
+                self.restore_original_state()
+                self.module.fail_json(msg=task.info.error.msg)
+
+    def remove_vspan_session(self):
+        """Calls the necessary functions to delete a VSpanSession."""
+        results = dict(changed=False, result="")
+        mirror_session = self.find_session_by_name()
+        if mirror_session is None:
+            self.module.fail_json(msg='There is no VSpanSession with this name.')
+        promiscuous_ports = self.turn_off_promiscuous()
+        session_key = mirror_session.key
+        # Delete Mirroring Session
+        self.delete_mirroring_session(session_key)
+        # Session
+        self.deleted_session = mirror_session
+        # Set back the promiscuous ports
+        if promiscuous_ports:
+            for port in promiscuous_ports:
+                self.set_port_security_promiscuous(port, True)
+        results['changed'] = True
+        results['result'] = 'VSpan Session has been deleted'
+        return results
+
+    def check_if_session_name_is_free(self):
+        """Checks whether the name is used or not
+
+        Returns
+        -------
+        bool
+            True if the name is free and False if it is used.
+        """
+        for vspan_session in self.dv_switch.config.vspanSession:
+            if vspan_session.name == self.name:
+                return False
+        return True
+
+    def create_vspan_session(self):
+        """Builds up the session, adds the parameters that we specified, then creates it on the vSwitch"""
+
+        session = vim.dvs.VmwareDistributedVirtualSwitch.VspanSession(
+            name=self.name,
+            enabled=True
+        )
+        if self.session_type is not None:
+            session.sessionType = self.session_type
+            if self.session_type == 'encapsulatedRemoteMirrorSource':
+                if self.source_port_received is not None:
+                    port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(portKey=str(self.source_port_received))
```

Comment:
Suggest verifying the port exists, and fail if it doesn't (same for the other sections below)